extensions/gdx-freetype/src/com/badlogic/gdx/graphics/g2d/freetype/FreeTypeFontGenerator.java

```diff
@@ -801,5 +836,35 @@ public static class FreeTypeFontParameter {
 		 * modified after creating a font. If a PixmapPacker is not specified, the font glyph page textures will use
 		 * {@link FreeTypeFontGenerator#getMaxTextureSize()}. */
 		public boolean incremental;
+		/** Variable design coordinates. If the length exceeds the number of variable axes in the font, it will be ignored. If the
+		 * length is less than the number of variable axes in the font, the remaining coordinates will be set to their default
+		 * values. */
+		public int[] variableDesignCoordinates = new int[0];
+	}
+
+	public class VariableAxis {
+		private int minimum;
+		private int maximum;
+		private int def;
+
+		public VariableAxis (FreeType.VarAxis varAxis) {
+			this.minimum = (int)varAxis.getMinimum() / 65536;
+			this.maximum = (int)varAxis.getMaximum() / 65536;
+			this.def = (int)varAxis.getDef() / 65536;
```

Comment:
I believe this is not the right behavior some of the time, but I'm not sure. On the docs, https://freetype.sourceforge.net/freetype2/docs/reference/ft2-multiple_masters.html#FT_Var_Axis , minimum and maximum are FT_FIXED type, roughly equivalent to a float I think here, but on FreeType's side they're a signed `long` (it's C, so size is compiler- and platform-dependent). If it really is the 32-bit representation of a 16.16 fixed-point number, then the sign bit matters for how division works vs. a bitwise shift. The minimum and maximum shouldn't ever be negative conceptually, but they're being converted from a `long` (in Java), so I don't know if there could be a sign bit set as a result of that (if minimum or maximum is a large-enough positive `long`, then casting to `int` can produce a negative `int`).

Long story short, if you want `this.minimum` and `this.maximum` to be non-negative `int`s in the 0-65535 range, I would use `this.minimum = (int)varAxis.getMinimum() >>> 16;`, but if negative min and max are fine, then this can stay as-is.